src/net/gateway-archive.ts

```diff
@@ -0,0 +1,151 @@
+import { VochainProcessStatus } from "../models/protobuf";
+import { getEnsPublicResolverByNetwork } from "../util/ens";
+import { ContentUri } from "../wrappers/content-uri"
+import { EthNetworkID, IGatewayClient, IGatewayDVoteClient } from "../common";
+
+export interface IArchiveResponseBody {
+    process?: {
+        [k: string]: any
+    }
+    results?: {
+        [k: string]: any
+    },
+}
+
+export namespace GatewayArchive {
+
+    /**
+     * Fetch the full state of the given processId on the process archive
+     *
+     * @param processId
+     * @param gateway
+     * @param errorMessage
+     */
+    export function getProcessStateArchive(processId: string, gateway: IGatewayClient, errorMessage: string) {
+        return getProcess(processId, gateway, errorMessage)
+    }
+
+    /**
+     * Fetch the headers of the given processId on the archive.
+     *
+     * @param processId
+     * @param gateway
+     * @param errorMessage
+     */
+    export function getProcessSummaryArchive(processId: string, gateway: IGatewayClient, errorMessage: string) {
+        return getProcess(processId, gateway, errorMessage)
+            .then((response) => {
+                response.process.envelopeHeight = response.results.envelopeHeight
+                return {
+                    processSummary: response.process
+                }
+            })
+    }
+
+    /**
+     * Retrieves the archive cumulative weight that has been casted in votes for the given process ID.
+     *
+     * @param processId
+     * @param gateway
+     * @param errorMessage
+     */
+    export function getResultsWeightArchive(processId: string, gateway: IGatewayClient, errorMessage: string) {
+        return getProcess(processId, gateway, errorMessage)
+            .then((response) => {
+                return {
+                    weight: response.results.weight
+                }
+            })
+    }
+
+    /**
+     * Fetches the archive number of vote envelopes for a given processId
+     *
+     * @param processId
+     * @param gateway
+     * @param errorMessage
+     */
+    export function getEnvelopeHeightArchive(processId: string, gateway: IGatewayClient, errorMessage: string) {
+        return getProcess(processId, gateway, errorMessage)
+            .then((response) => {
+                return {
+                    height: response.results.envelopeHeight
+                }
+            })
+    }
+
+    /**
+     * Fetches the archive results for a given processId
+     *
+     * @param processId
+     * @param gateway
+     * @param errorMessage
+     * @returns Results, vote process  type, vote process state
+     */
+    export function getRawResultsArchive(processId: string, gateway: IGatewayClient, errorMessage: string) {
+        return getProcess(processId, gateway, errorMessage)
+            .then((response) => {
+                return {
+                    results: response.results.votes,
```

Comment:
Shouldn't this function be returning just the votes... instead of everything else?

Reply:
But the caller of this function `getRawResults` is expecting those values in `voting.ts`